File: src/main/java/com/vicen/films/framework/presentation/rest/BaseController.java
package com.vicen.films.framework.presentation.rest;

import com.vicen.films.framework.errors.ApplicationError;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;

import javax.servlet.http.HttpServletRequest;

public abstract class BaseController {

    @ExceptionHandler(ApplicationError.class)
    public ResponseEntity handleError(HttpServletRequest req, ApplicationError ex) {
        return new ResponseEntity(ex.getStatus());
    }

}

File: src/main/kotlin/com/vicen/films/framework/presentation/rest/BaseController.kt
package com.vicen.films.framework.presentation.rest

import com.vicen.films.framework.errors.ApplicationError
import com.vicen.films.framework.errors.BadRequestError
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import javax.servlet.http.HttpServletRequest

abstract class BaseController {

    @ExceptionHandler(ApplicationError::class)
    fun handleError(req: HttpServletRequest, ex: ApplicationError) = ResponseEntity<Any>(ex.status)

    protected fun <T> throwMandatoryFieldError(fieldName : String) : T {
        throw BadRequestError("Field $fieldName is mandatory.")
    }

}